### ISSUE TITLE
Enhancement: Dropdown menu to select OpenCL version

### ISSUE THREAD

**Issue by TSlivede** [KhronosGroup/OpenCL-Docs#305]
As already briefly noted [here](https://github.com/KhronosGroup/OpenCL-Docs/issues/251#issuecomment-633670561):

Is it possible, to add a drop-down-menu to the header of all reference pages (old ones and new ones), that lets one switch to a newer or older version?

Alternatively, could at least a link to the latest version be added to all older versions?

**Comment by oddhack** [comment]
Responding to https://github.com/KhronosGroup/OpenCL-Docs/issues/251#issuecomment-634189106 in the parent issue: doing this isn't really an asciidoctor issue. The navigation setup is semi-static - when the individual HTML refpages are generated from the spec, it writes an index of pages that's combined into toc.html (see the files that are checked in under man/, they are the static portion of the refpages). There are also some static pages (not generated from the spec) under man/static/ that are combined in the same way.

The toolchains for the 2.2 refpages is totally different than the older APIs, which are Docbook XML -> XHTML and use classic HTML \<frame>s for navigation. I wouldn't want to modify all the individual pages to add cross-links as that would mean making changes to thousands of documents and regenerating all the 2.1 and earlier pages. It is tractable to add something to the top-level navigation in each API version's pages to switch back and forth, as those would just be crosslinks between the refpage sets. But that would not be the requested functionality for switching back and forth between different API versions of the same refpage.

A different angle on this would be to instrument the current refpages to document which API version a given page was introduced in. That is automatable although we'd have to add some tagging in the spec markup or in cl.xml and modify the refpage extraction script to use that. We do something like that with the current GL refpages, in a table at the end of each one - see for example "Version Support" under https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glBindSamplers.xhtml .

**Comment by oddhack** [comment]
@TSilvede correct me if I'm misreading, but I think you're after something like https://docs.python.org/2.7/library/index.html where there's a dropdown menu at the top of each page in the docs selecting versions.

**Comment by TSlivede** [comment]
@oddhack 
> [...] I think you're after something like https://docs.python.org/2.7/library/index.html where there's a dropdown menu at the top of each page in the docs selecting versions.

Yes, that's what I had in mind, sorry if I didn't explain clearly enough, I'm not a native English speaker :-)

---

> A different angle on this would be to instrument the current refpages to document which API version a given page was introduced in.

Yes, that would be very nice!

---

If additionally a plain link to the latest refpages could be added to the old ones, that would completely fulfill the purpose. But I don't know, is the addition of a plain link (to the old refpages) easier than a dropdown menu?

**Comment by TSlivede** [comment]
>I wouldn't want to modify all the individual pages to add cross-links as that would mean making changes to thousands of documents and regenerating all the 2.1 and earlier pages.

I don't think, it would be appropriate to add a dropdown menu or 'link to latest' to all *individual* pages. That would mean, that the link/dropdown menu appears beneath the actual header if one visits a refpage through the [main page](https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/) (because it is displayed within an frame).

Instead such a 'link to latest'/dropdown-menu should just be added into these pages, I think:
https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/Opencl_header.html    
https://www.khronos.org/registry/OpenCL/sdk/1.1/docs/man/xhtml/Opencl_header.html    
https://www.khronos.org/registry/OpenCL/sdk/1.2/docs/man/xhtml/Opencl_header.html    
https://www.khronos.org/registry/OpenCL/sdk/2.0/docs/man/xhtml/Opencl_header.html    
https://www.khronos.org/registry/OpenCL/sdk/2.1/docs/man/xhtml/Opencl_header.html    

In case Javascript is available/enabled in the browser, that link could submit (in the query or fragment part of the URI) the page that is currently displayed in the main frame, such that the main page of the current refpages directly opens the corresponding current refpage in the frame...

**Comment by TSlivede** [comment]
Just for thinking process: In the current refpages https://github.com/KhronosGroup/OpenCL-Docs/blob/master/man/index.html seems to be a static (not generated) page, that could be easily changed. Are the index.html files of the old refpages (https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/index.html, etc.) also static (not generated) pages, that could be easily changed?
